walletconnectv2/src/main/kotlin/org/walletconnect/walletconnectv2/engine/serailising/JsonRpcSerializer.kt

```diff
@@ -0,0 +1,17 @@
+package org.walletconnect.walletconnectv2.engine.serailising
+
+import com.squareup.moshi.JsonDataException
+import com.squareup.moshi.Moshi
+
+class JsonRpcSerializer(val moshi: Moshi) : JsonRpcSerialising {
+
+    override fun <T> trySerialize(typeClass: Class<T>, type: T): String =
+        moshi.adapter(typeClass).toJson(type)
```

Comment:
Similar comment as below. Using `reified` will simply the function signature
```suggestion
    inline fun <reified T> trySerialize(type: T): String =
        moshi.adapter(T::class.java).toJson(type)
```